src/events/index.ts

```diff
@@ -0,0 +1,11 @@
+import { ExtensionContext } from 'vscode';
```

Comment:
I'm still ramping up here, but from what I can tell so far this "Events" pattern is not a good one. In most (all?) cases there isn't a need to leverage VSCode's event bus to execute logic like this. AFAIK the primary reason to register events is to expose them as user-facing functionality in the `contributes` section of the package.json. These events are adding unnecessary complexity to the logic flow by leveraging the event bus when they don't need to be.

I undid some of it (all the terminal logic that checks the installation status for npm and the cli, and also the global state events), but not all of it because some events are pretty deeply rooted.